mayastor/src/pool.rs

```diff
@@ -560,6 +583,63 @@ pub(crate) async fn create_pool(
     Ok(pool.into())
 }
 
+fn is_uri_scheme(disks: &[String]) -> bool {
+    !disks.iter().any(|d| Url::parse(d).is_err())
+}
+
+async fn create_pool_uri(args: rpc::CreatePoolRequest) -> Result<rpc::Pool> {
+    if args.disks.len() != 1 {
+        return Err(Error::BadNumDisks {
+            num: args.disks.len(),
+        });
+    }
+
+    let parsed = Uri::parse(&args.disks[0]).map_err(|e| Error::BadBdev {
+        bdev_if: e.to_string(),
+        name: args.disks[0].clone(),
+    })?;
+
+    if let Some(pool) = Pool::lookup(&args.name) {
+        return if pool.get_base_bdev().name() == parsed.get_name() {
+            Ok(pool.into())
+        } else {
+            Err(Error::AlreadyExists {
+                name: args.name,
```

Comment:
should this error include the existing disk as well?

Reply:
Likely yes, but all these error variants, all get smooshed down into the same error anyway (internal or not found) so I'd like to address that in the follow-up releases were we reduce the error variants.